test/shared/src/main/scala/zio/test/TestRunner.scala

```diff
@@ -37,7 +37,7 @@ case class TestRunner[L, -T](
   /**
    * Runs the spec, producing the execution results.
    */
-  final def run(spec: Spec[L, T]): URIO[TestLogger with Clock, ExecutedSpec[L]] =
+  final def run(spec: Spec[L, T]): URIO[TestLogger with Clock, ExecutedSpec[L, E, S]] =
```

Comment:
One thing disturbs me about this: Where are these `E` and `S` values coming from? Of course we know they are in `T` but the code does not know that. I think we can still merge and figure it out later.